sentry-kotlin-multiplatform/src/commonTest/kotlin/io/sentry/kotlin/multiplatform/SentryIntegrationTest.kt

```diff
@@ -0,0 +1,266 @@
+        var actualLevel: SentryLevel? = null
+        sentryInit {
+            it.dsn = fakeDsn
+            it.beforeSend = { event ->
+                actualLevel = event.level
+                null
+            }
+        }
+
+        Sentry.configureScope {
+            it.level = expectedLevel
+        }
+
+        Sentry.captureException(RuntimeException("test"))
+
+        assertNotNull(actualLevel)
+        assertEquals(expectedLevel, actualLevel)
+    }
+
+    @Test
+    fun `global scope sets user correctly`() {
+        val expectedEmail = "test@example.com"
+        val expectedId = "123"
+        val expectedIpAddress = "127.0.0.1"
+        val expectedUsername = "testuser"
+        var actualEmail = ""
+        var actualId = ""
+        var actualIpAddress = ""
+        var actualUsername = ""
+
+        sentryInit {
+            it.dsn = fakeDsn
+            it.beforeSend = { event ->
+                val user = event.user
+                actualEmail = user?.email ?: ""
+                actualId = user?.id ?: ""
+                actualIpAddress = user?.ipAddress ?: ""
+                actualUsername = user?.username ?: ""
+                null
+            }
+        }
+
+        Sentry.configureScope {
+            it.user = User().apply {
+                this.email = expectedEmail
+                this.id = expectedId
+                this.ipAddress = expectedIpAddress
+                this.username = expectedUsername
+            }
+        }
+
+        Sentry.captureException(RuntimeException("test"))
+
+        assertEquals(expectedEmail, actualEmail)
+        assertEquals(expectedId, actualId)
+        assertEquals(expectedIpAddress, actualIpAddress)
+        assertEquals(expectedUsername, actualUsername)
+    }
+
+    @Test
+    fun `global scope sets context correctly with different data types`() = runTest {
+        val stringKey = "stringKey"
+        val stringValue = "stringValue"
+        val booleanKey = "booleanKey"
+        val booleanValue = true
+        val numberKey = "numberKey"
+        val numberValue = 123
+        val collectionKey = "collectionKey"
+        val collectionValue = listOf("abc", 123, true)
+
+        val deferred = CompletableDeferred<Unit>()
+        sentryInit {
+            it.dsn = fakeDsn
+            it.beforeSend = { event ->
+                val contexts = event.contexts
+                try {
+                    assertNotNull(contexts)
+                    assertEquals(mapOf("value" to stringValue), contexts[stringKey])
+                    assertEquals(mapOf("value" to booleanValue), contexts[booleanKey])
+                    assertEquals(mapOf("value" to numberValue), contexts[numberKey])
+                    assertEquals(mapOf("value" to collectionValue), contexts[collectionKey])
+                    deferred.complete(Unit)
+                } catch (e: Throwable) {
+                    deferred.completeExceptionally(e)
+                }
```

Comment:
We can assert after, no need of try-catch, otherwise, if the assert fails, the test fails anyway, or is there any other reason?